greykite/algo/forecast/silverkite/forecast_silverkite.py

```diff
@@ -2678,6 +2677,9 @@ def partition_fut_df(
             # otherwise it causes error with pandas>=1.4.
             fut_df_expanded.loc[:, fut_df_expanded.columns != time_col] = na_fill_func(
                 fut_df_expanded.loc[:, fut_df_expanded.columns != time_col])
+            # Turn 'bool' columns from dtype object to bool
+            bool_col_names = [col for col in fut_df_expanded.columns if 'bool' in col]
```

Comment:
Obecně nerad používám prostou kontrolu `'pattern' in str`, protože to může vést k neočekávaným falešným pozitivům. Z kódu se mi nedaří vyčíst strukturu dataframe, ale pokud mají nějakou pevně danou strukturu, např. prefix/suffix, tak by bylo lepší použít kontrolu ve stylu `col.endswith('_bool')`, `col.startswith('bool_')` či něco podobného.

Reply:
Vidět je to v odpovídajícím souboru s testy. Je tam třeba sloupec `'regressor_bool', 'regressor_bool_lag168',        'regressor_bool_avglag_168_336_504'`. V anglickém slovníku je jako slova obsahující `'bool'` jenom tabooley, babool.
Zúžila jsem to na `'_bool'`.

Reply:
@Shavril Určitě, takhle je to lepší. Ono nejde jen o slova, ale o použití v jiném kontextu. Viděl jsem už i třeba sloupce `bool_count`, který obsahoval počet platných hodnot typu `bool` v nějakém datasetu. Čím specifičtější máme podmínku, tím lepší, i když třeba nebude dokonalá.